### PR TITLE
fix: handle escaped script data end tags

### PR DESCRIPTION
**Pull request by harlan-zw**
### 🔗 Linked issue

Related to #132

Stacked on #137.

### ❓ Type of change

- [ ] 📖 Documentation
- [x] 🐞 Bug fix
- [ ] 👌 Enhancement
- [ ] ✨ New feature
- [ ] 🧹 Chore
- [ ] ⚠️ Breaking change

### 📚 Description

Legacy `<!-- ... -->` wrappers can put `<script>` content into escaped or double-escaped tokenizer states. The JavaScript and Rust parsers now follow the boundary-relevant script-data states before accepting `</script>`, with Chromium and html5lib cases covering malformed names, mixed case, EOF, and every two-chunk split.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved streamed parsing of unnamed `<script>` bodies to reliably find the correct first `</script>` closing boundary, including escaped and double-escaped variants.
  * Correctly buffers and defers parsing when the closing boundary is split across chunks, and emits the expected trailing content when a `<script>` is left unterminated.
  * Keeps script extraction and rendered output consistent across arbitrary stream chunking.

* **Tests**
  * Added regression and streaming test coverage for multiple escaping/end-tag edge cases, extraction correctness, and unterminated `<script>` remainder handling.

### CODE REVIEW

**Comment by harlan-zw** [comment]
@coderabbitai review

File: crates/core/src/convert/mod.rs [hunk 1]
```diff
@@ -354,6 +527,27 @@ impl ConvertState {
...
+        .stack
+        .last()
+        .is_some_and(|node| node.tag_id == Some(TAG_SCRIPT) && node.custom_name.is_none())
+      {
+        let Some(close_index) = find_script_end_tag(bytes, i) else {
+          text_buffer.push_str(&chunk[i..]);
+          self.text_buffer_contains_non_whitespace = true;
+          self.last_char_was_whitespace = false;
+          self.just_closed_tag = false;
+          break;
```
[the remark this answers is not in the document]
Comment (harlan-zw) [reply crates/core/src/convert/mod.rs]: Fixed in 2f28fad. Both engines now persist the script tokenizer state and return only an incomplete boundary candidate. Consumed script text is retained separately for extraction/plugin events. A one-character-chunk benchmark improved from 223 ms to 5 ms in JavaScript and from 11 ms to 4 ms in Rust; full tests, clippy, lint, typecheck, and build pass.

**Comment by harlan-zw** [comment]
@coderabbitai review
